server/src/test/java/io/crate/planner/CollectPlannerTest.java

```diff
@@ -0,0 +1,62 @@
+/*
+ * Licensed to Crate.io GmbH ("Crate") under one or more contributor
+ * license agreements.  See the NOTICE file distributed with this work for
+ * additional information regarding copyright ownership.  Crate licenses
+ * this file to you under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.  You may
+ * obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.  See the
+ * License for the specific language governing permissions and limitations
+ * under the License.
+ *
+ * However, if you have executed another commercial license agreement
+ * with Crate these terms will supersede the license and you may use the
+ * software solely pursuant to the terms of the relevant commercial agreement.
+ */
+
+package io.crate.planner;
+
+
+import static org.assertj.core.api.Assertions.assertThat;
+
+import java.util.List;
+
+import org.junit.Test;
+
+import io.crate.analyze.WhereClause;
+import io.crate.analyze.relations.DocTableRelation;
+import io.crate.expression.symbol.Symbol;
+import io.crate.metadata.doc.DocTableInfo;
+import io.crate.planner.operators.Collect;
+import io.crate.planner.operators.LogicalPlan;
+import io.crate.test.integration.CrateDummyClusterServiceUnitTest;
+import io.crate.testing.SQLExecutor;
+
+public class CollectPlannerTest extends CrateDummyClusterServiceUnitTest {
+
+    /*
+     * https://github.com/crate/crate/issues/16047
+     */
+    @Test
+    public void test_pruning_works_on_scalars() throws Exception {
+        SQLExecutor e = SQLExecutor.of(clusterService)
+            .addTable("create table tbl (a int, b int)");
+        DocTableInfo tbl = e.resolveTableInfo("tbl");
+        var tableRelation = new DocTableRelation(tbl);
+
+        Symbol a = e.asSymbol("a");
+        Symbol b = e.asSymbol("b");
+
+        Symbol function = e.asSymbol("case when a is not null then a else b end");
+
+        Collect collect = new Collect(tableRelation, List.of(a, b), WhereClause.MATCH_ALL);
+
+        LogicalPlan pruned = collect.pruneOutputsExcept(List.of(function));
+        assertThat(pruned.outputs()).containsExactly(b, a);
+    }
```

Comment:
Do I get it right that inverted order is not related to this PR but rather  to Case function handling outputs from the end/else?